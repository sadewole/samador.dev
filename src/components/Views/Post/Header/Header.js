import React from "react"
import PropTypes from "prop-types"
import { Link } from "gatsby"
import ToggleTheme from "../../../toggle"

const Header = ({ author }) => {
  return (
    <nav className="mb-5 flex items-center justify-between">
      <h3>
        <Link to="/blogs">{author.name}'s blog</Link>
      </h3>
      <ToggleTheme />
    </nav>
  )
}

Header.propTypes = {
  author: PropTypes.object,
}

export default Header
