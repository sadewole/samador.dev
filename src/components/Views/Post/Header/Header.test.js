import React from "react"
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"

import Header from "./Header"

describe("Header", () => {
  const props = {
    author: { name: "test" },
  }

  it("must contain string", () => {
    render(<Header {...props} />)
    const name = `${props.author.name}'s blog`

    expect(screen.getByTestId("blog-nav")).toHaveTextContent(name)
  })

  it("render correctly", () => {
    const tree = renderer.create(<Header {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
