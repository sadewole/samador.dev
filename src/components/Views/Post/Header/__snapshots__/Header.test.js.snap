// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render correctly 1`] = `
<nav
  className="mb-5 flex items-center justify-between"
>
  <h3
    data-testid="blog-nav"
  >
    <a
      href="/blogs"
    >
      test
      's blog
    </a>
  </h3>
  <button
    aria-hidden="true"
    className={null}
    onClick={[Function]}
    style={
      Object {
        "MozAppearance": "none",
        "WebkitAppearance": "none",
        "appearance": "none",
        "backgroundColor": "transparent",
        "border": "none",
        "cursor": "pointer",
        "height": "37.599999999999994px",
        "overflow": "hidden",
        "padding": 0,
        "width": "80px",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "132px",
          "justifyContent": "center",
          "marginLeft": "-25.6px",
          "marginTop": "-47.599999999999994px",
          "width": "132px",
        }
      }
    >
      <div
        aria-label="animation"
        role="button"
        style={
          Object {
            "height": "100%",
            "margin": "0 auto",
            "outline": "none",
            "overflow": "hidden",
            "width": "100%",
          }
        }
        tabIndex={0}
        title=""
      />
    </div>
  </button>
</nav>
`;
